### DISCUSSION TITLE
How to redirect sql_exporter log to required file

### DISCUSSION THREAD

**Discussion by ignalexeyp** [burningalchemist/sql_exporter#637] [Q&A] [answered]
Is it possible to redirect the sql_exporter log to the required file (for example: /var/lib/postgresql/sqlexplog/sql_exporter.log)? Currently, the sql_exporter log is written to the system log /var/log/syslog.
The sql_exporter.yml settings are according to the instructions https://github.com/burningalchemist/sql_exporter/blob/master/documentation/sql_exporter.yml.
I did not change the settings in sql_exporter.service.

sql_exporter, version 0.16.0 (branch: HEAD, revision: a1726ff66775980c1de3ad44bfffb169525b2dd1)
  build user:       root@214dc5fa2322
  build date:       20241026-08:34:31
  go version:       go1.23.2
  platform:         linux/amd64
  tags:             netgo

No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 22.04.5 LTS
Release:        22.04
Codename:       jammy

PostgreSQL 16.3 (Ubuntu 16.3-1.pgdg22.04+1)

**Comment by burningalchemist** [comment, marked as answer]
Hi @ignalexeyp, I've recently switched to another logging library (`log/slog`), and it's possible now to define the log filepath to write to (by default it's stdout).

The feature is already available on `master`, but wasn't released yet as a part of `v0.17`.

So at the moment there are three options:
- you might build a release yourself from master (it's straightforward - just follow the docs);
- wait a little bit before the new release is cut;
- and you can also configure syslog to redirect stdout data to the file in the meantime.

**Reply by ignalexeyp** [reply]
Thanks for the recommendations. I'll try them.
